src/main/java/com/datastax/logging/appender/CassandraAppender.java

```
@@ -238,507 +85,315 @@ private void initTransportFactory()
-            createMutation(mutList, THROWABLE_STR, builder.toString(), colTs);
-        }
-        createMutation(mutList, TIMESTAMP, event.getTimeStamp(), colTs);
+            String[] contactPoints = hosts.split(",\\s*");
 
-        return mutList;
-    }
+            cluster = Cluster.builder()
+						  .addContactPoints(contactPoints)
+						  .withPort(port)
+						  .withCredentials(username, password)
+//                          .withLoadBalancingPolicy(new DCAwareRoundRobinPolicy("Solr"))
+                          .withLoadBalancingPolicy(new RoundRobinPolicy())
+                          .build();
 
-    private void createMutation(List<Mutation> mutList, String column, long value, long ts)
-    {
-        createMutation(mutList, column, toByteBuffer(value), ts);
-    }
+            // FIXME: we'd like to use DCAwareRoundRobinPolicy here, but we need to figure out the datacenter
+            // first.  Probably this should be a configuration option (although the question is where to put it)
+            // but it should also be possible to get some very reasonable default value regardless.  In fact, the
+            // cluster itself does know, but we'd have to setup and use it twice.
 
-    private void createMutation(List<Mutation> mutList, String column, String value, long ts)
-    {
-        if (value != null)
+		    session = cluster.connect();
+            setupSchema();
+            setupStatement();
+		}
+        catch (Exception e)
         {
-            createMutation(mutList, column, toByteBuffer(value), ts);
-        }
-    }
+		    LogLog.error("Error ", e);
+			errorHandler.error("Error setting up cassandra logging schema: " + e);
+
+            //If the user misconfigures the port or something, don't keep failing.
+            initializationFailed = true;
+		}
+        finally
+        {
+            //Always reenable logging
+            LogManager.getLoggerRepository().setThreshold(globalThreshold);
+		}
+
+        initialized = true;
```

I appreciate that this isn't a new thing here, but the handling of errors during setup is a bit off. For instance, if we fail to init the client for whatever reason, we swallow the exception and so continue trying to execute the prepared statement. 
Also, because we set initialized=true regardless, next time around we'll attempt & fail the query again. 

The alternatives would seem to be either retry the initialization each time we try to append (which would be pointless if the failure is down to a misconfiguration), or to simply log the first failure then make further append calls no-op if initializationFailed=true.

wdyt?